### PR TITLE
[Weather] Add weather datasets

### PR DESCRIPTION
**Pull request by RattataKing**
- Add processed data for 4 climate stations in `climate_data` folder
  - `TORONTO_CITY`
  - `TORONTO_CITY_CENTRE`
  - `TORONTO_INTL_A`
  - `TORONTO_NORTH_YORK`
  
- Add scripts in `data_processing_script` folder:
  - `cat.py`
  - `basic_clean.py`
  - `deep_clean.py`
  - `extract_neighborhood.py`
  - `process_data.py`
  - `compare_pcc.py`
  - `download_weather_data.sh`
  
- Add README:
  - main `README.md`
  - `HISTORICAL_CLIMATE_DATA_INFO.md` under `climate_data` folder

### CODE REVIEW

**Comment by RattataKing** [comment]
@LindyZh need further advices

File: Amily_weather/Climate Data/README.md
Comment (LindyZh) [inline Amily_weather/Climate Data/README.md:1]: can we rename this to differentiate from the main README?

File: Amily_weather/Climate Data/TORONTO_CITY/Cleaned/basic_clean_data.csv
Comment (LindyZh) [inline Amily_weather/Climate Data/TORONTO_CITY/Cleaned/basic_clean_data.csv:1]: **TODO**: for these types of the larger csvs, can we migrate this to google drive? 

Right now there is 230k code additions, we can add a link to these codes in the README.md where later we can use google drive to mount the data. 

so the folder structure should look something like:

.
└── weather/
    ├── climate_data/
    │   ├── pcc_matrix.csv
    │   ├── README.md (rename this so it is differentiated from the main readme)
    │   └── STATION_A/
    │       ├── report
    │       └── (other large csv migrate to google drive)
    ├── data_processing_script/
    │   └── python scripts...let me know if there is codes that we want to recycle
    └── README.md

Comment (LindyZh) [reply Amily_weather/Climate Data/TORONTO_CITY/Cleaned/basic_clean_data.csv:1]: I was also not too sure about the difference between the basic_clean/deep_clean/filled_deep_clean (although I kinda know). I think there's 2 ways to go about this:
1. put this work in process in a cache.zip and put it on the drive, and update the folder structure to only keep the 4 `processed_data.csv` with station prefix (this is still large)
2. use a zip to preserve the processed_data.csv in google drive, put all the other in a cache zip in local/drive, but link to both in main README

File: Amily_weather/extract_neighborhood.py
Comment (LindyZh) [inline Amily_weather/extract_neighborhood.py:1]: is this script still valid after we decided to go with the police dataset?

File: Amily_weather/toronto_neighborhood.csv
Comment (LindyZh) [inline Amily_weather/toronto_neighborhood.csv:1]: instead of having this here, maybe just put this in main

**Review by LindyZh — APPROVED** [review]
I left a few comments regarding optimizing the data volumes, let me know what you think 🥇

File: Amily_weather/extract_neighborhood.py
Comment (RattataKing) [reply Amily_weather/extract_neighborhood.py:1]: this is the a temporary solution when we don't have an unified comprehensive neighbourhood list, will add one in main instead.

File: Amily_weather/Climate Data/README.md
Comment (RattataKing) [reply Amily_weather/Climate Data/README.md:1]: pick one for me

File: Amily_weather/extract_neighborhood.py
Comment (LindyZh) [reply Amily_weather/extract_neighborhood.py:1]: I see, I think we could take this out from the current pr for now, I'll add a new issue regarding fixing the neighbourhood files

File: Amily_weather/Climate Data/TORONTO_CITY/Cleaned/basic_clean_data.csv
Comment (RattataKing) [reply Amily_weather/Climate Data/TORONTO_CITY/Cleaned/basic_clean_data.csv:1]: done

File: Amily_weather/Climate Data/README.md
Comment (RattataKing) [reply Amily_weather/Climate Data/README.md:1]: decide to use `HISTORICAL_CLIMATE_DATA_INFO.md`, but I moved the most of important parts from the sub-README to the main-README, now it becomes a little bit unnecessary now, you decide whether we still want `HISTORICAL_CLIMATE_DATA_INFO.md`

**Comment by RattataKing** [comment]
before merge, need to regenerate `*_processed_data.csv` with the newest `neighbourhood_data.csv` from main, and update `process_data.py` and `README.md`

File: Amily_weather/Climate Data/README.md
Comment (RattataKing) [reply Amily_weather/Climate Data/README.md:1]: will keep `HISTORICAL_CLIMATE_DATA_INFO.md`

**Comment by RattataKing** [comment]
moved all work to upstream repo branch (to easier merge into upstream/main), see #4

**Pull request closed without merging**